macros/parsers/parserLinearRelation.pl

```diff
@@ -0,0 +1,379 @@
+		},
+		'<=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' <= ',
+			TeX           => '\le ',
+			kind          => "le",
+			eval          => sub { $_[0] <= $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => [ '=<', "\x{2264}" ]
+		},
+		'>=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' >= ',
+			TeX           => '\ge ',
+			kind          => 'ge',
+			reverse       => 'le',
+			eval          => sub { $_[0] >= $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => [ '=>', "\x{2265}" ]
+		},
+		'!=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' != ',
+			TeX           => '\ne ',
+			kind          => 'ne',
+			eval          => sub { $_[0] != $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => [ '<>', "\x{2260}" ]
+		},
+	);
+	main::PG_restricted_eval('sub LinearRelation {Value->Package("LinearRelation()")->new(@_)}');
+}
+
+#    $R = LinearRelation("x + y + 2z <= 5");
+#    $R = Formula("x + y + 2z <= 5");
+#    $R = Compute("x + y + 2z <= 5");
+#    $R = LinearRelation([1,2,1], [1,1,2], "<=");
+#    $R = LinearRelation([1,2,1], 5, "<=");
+
+sub new {
+	my ($self, $N, $p, $bop) = @_;
+	my $class   = ref($self) || $self;
+	my $context = (Value::isContext($_[0]) ? shift : $self->context);
+	my $formula = "Value::Formula";
+	return shift if scalar(@_) == 1 && ref($_[0]) eq 'LinearRelation';
+	#$_[0] = $context->Package("Vector")->new($context, $_[0]) if ref($_[0]) eq 'ARRAY';
+	#$_[1] = $context->Package("Point")->new($context, $_[1])  if ref($_[1]) eq 'ARRAY';
+
+	# 2 or more:
+	# first argument is normal vector, possibly a Vector, Point, or ARRAY ref
+	# second argument is point or number, possibly a Vector, Point, ARRAY ref, Real, or perl real
+	# optional third argument is symbol, defaults to '='
+
+	my ($plane, $d);
+
+	if (defined $p) {
+		# Make sure $N is a Vector
+		$N = $context->Package("Vector")->new($context, $N);
+		# Make sure $p is a Point or Real
+		$p = $context->Package("Point")->new($context, $p) if (ref($p) eq 'ARRAY' || ref($p) eq 'Value::Vector');
+		$p = $context->Package("Real")->new($context, $p)  if ref($p) ne 'Value::Point';
+		# Constant on the right side
+		$d = (ref($p) eq 'Value::Real') ? $p : $p . $N;
+		my @terms;
+		my $i = 0;
+		for my $x ($context->variables->names) { push @terms, $N->{data}[ $i++ ]->string . $x }
+		$bop      = '=' unless defined($bop);
+		$leftside = $formula->new(join(' + ', @terms))->reduce(@_);
+		$plane    = $formula->new($leftside . $bop . $d->string);
+	} else {
+		# Determine the normal vector and d value from the equation
+		$plane = $N;
+		$context->flags->set(reduceConstants => 0);
```

Comment:
I don't think this is the best way to accomplish your goal of keeping things like `0 < 1` as a formula, as this setting is permanent on the context, and so will prevent any constant reduction in any computation within that context.

Instead, you could make the inequality operators always mark themselves as non-constant, so they are never reduced by adding

``` perl
        $self->{isConstant} = 0;
```

at the end of the `_check` function of `LinearRelation::inequality`.  This will make `LinearRelation("0 < 1")` remain as a formula "0 < 1".  However, `Compute("0 < 1")` will still return `1`, since it uses `$formula->isConstant` to determine whether to reduce the final result, and that function tests whether there are any variables in the formula to tell if it is constant rather than the `isConstant` property of the top node in the tree. 

To handle that case as well, you can add

``` perl
sub isConstant {
  my $self = shift;
  return 0 if $self->type eq "Relation";
  return self->SUPER::isConstant;
}
```

to the end of the `LinearRelation` class.  Then `Compute("0 < 1")` will return a Formula rather than reducing it a constant.

Using this approach will allow `reduceConstants` to act normally on everything other than relations, which are never reduced.